File: JavaScript/All-Inclusive.js
const containAllRots = (strng, arr) =>
  [...strng]
    .map((_, idx) => strng.slice(idx) + strng.slice(0, idx))
    .every((rot) => arr.includes(rot));

// ("bsjq", ["bsjq", "qbsj", "sjqb", "twZNsslC", "jqbs"]) -> true)
// ("Ajylvpy", ["Ajylvpy", "ylvpyAj", "jylvpyA", "lvpyAjy", "pyAjylv", "vpyAjyl", "ipywee"]) -> false)

File: Python/All-Inclusive.py
def contain_all_rots(strng, arr):
    arr_set = set(arr)

    for i in range(len(strng)):
        rotation = strng[i:] + strng[:i]
        if rotation not in arr_set:
            return False

    return True

# ("bsjq", ["bsjq", "qbsj", "sjqb", "twZNsslC", "jqbs"]) -> true)
# ("Ajylvpy", ["Ajylvpy", "ylvpyAj", "jylvpyA", "lvpyAjy", "pyAjylv", "vpyAjyl", "ipywee"]) -> false)
